lib/recent_test.go

```diff
@@ -0,0 +1,150 @@
+package lib
+
+import (
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+)
+
+func Test_convertData(t *testing.T) {
+	recentFileContent := []byte("1.5.6\n0.13.0-rc1\n1.0.11\n")
+
+	var recentFileData RecentFiles
+	convertOldRecentFile(recentFileContent, &recentFileData)
+	assert.Equal(t, 3, len(recentFileData.Terraform))
+	assert.Equal(t, 0, len(recentFileData.OpenTofu))
+	assert.Equal(t, "1.5.6", recentFileData.Terraform[0])
+	assert.Equal(t, "0.13.0-rc1", recentFileData.Terraform[1])
+	assert.Equal(t, "1.0.11", recentFileData.Terraform[2])
+}
+
+func Test_saveFile(t *testing.T) {
+	var recentFileData = RecentFiles{
+		Terraform: []string{"1.2.3", "4.5.6"},
+		OpenTofu:  []string{"6.6.6"},
+	}
+	temp, err := os.MkdirTemp("", "recent-test")
+	if err != nil {
+		t.Errorf("Could not create temporary directory")
+	}
+	defer func(path string) {
```

Comment:
I'm not 100% sure, but I think that simply
```
  defer os.RemoveAll(temp)
```
would provide the same functionality

Reply:
It would but would not ignore errors explicitely. At least my IDE states thats.